src/library_exceptions.js

```diff
@@ -7,92 +7,196 @@
 var LibraryExceptions = {
   __exception_last: '0',
   __exception_caught: ' []',
-  __exception_infos: '{}',
-
-  __exception_deAdjust__deps: ['__exception_infos'],
-  __exception_deAdjust: function(adjusted) {
-    if (!adjusted || ___exception_infos[adjusted]) return adjusted;
-    for (var key in ___exception_infos) {
-      var ptr = +key; // the iteration key is a string, and if we throw this, it must be an integer as that is what we look for
-      var adj = ___exception_infos[ptr].adjusted;
-      var len = adj.length;
-      for (var i = 0; i < len; i++) {
-        if (adj[i] === adjusted) {
-#if EXCEPTION_DEBUG
-          err('de-adjusted exception ptr ' + adjusted + ' to ' + ptr);
+
+  __ExceptionInfoAttrs: {
```

Comment:
please add a comment here explaining what an ExceptionInfo is in general, and maybe some notes on where it's allocated and freed, etc. - the big picture of ExceptionInfo and CatchInfo and their relationship would be good to describe.